solidity/contracts/BancorNetwork.sol

```diff
@@ -176,7 +237,86 @@ contract BancorNetwork is IBancorNetwork, TokenHolder, ContractIds, FeatureIds {
         if (msg.value > 0)
             IEtherToken(fromToken).deposit.value(msg.value)();
 
-        return convertForInternal(_path, _amount, _minReturn, _for, _block, _v, _r, _s);
+        return convertForInternal(_path, _amount, _minReturn, _for, _customVal, _block, _v, _r, _s);
+    }
+
+    /**
+        @dev converts any other token to BNT in the bancor network
+        by following a predefined conversion path and transfers the resulting
+        tokens to BancorX.
+        note that the network should already have been given allowance of the source token (if not ETH)
+
+        @param _path            conversion path, see conversion path format above
+        @param _amount          amount to convert from (in the initial source token)
+        @param _minReturn       if the conversion results in an amount smaller than the minimum return - it is cancelled, must be nonzero
+        @param _toBlockchain    blockchain BNT will be issued on
+        @param _to              address/account on _toBlockchain to send the BNT to
+        @param _xTransferId     pre-determined unique (if non zero) id which refers to this transaction 
+
+        @return the amount of BNT received from this conversion
+    */
+    function xConvert(
+        IERC20Token[] _path,
+        uint256 _amount,
+        uint256 _minReturn,
+        bytes32 _toBlockchain,
+        bytes32 _to,
+        uint256 _xTransferId
+    )
+        public
+        payable
+        returns (uint256)
+    {
+        return xConvertPrioritized(_path, _amount, _minReturn, _toBlockchain, _to, _xTransferId, 0x0, 0x0, 0x0, 0x0);
+    }
+
+    /**
+        @dev converts any other token to BNT in the bancor network
+        by following a predefined conversion path and transfers the resulting
+        tokens to BancorX.
+        this version of the function also allows the verified signer
+        to bypass the universal gas price limit.
+        note that the network should already have been given allowance of the source token (if not ETH)
+
+        @param _path            conversion path, see conversion path format above
+        @param _amount          amount to convert from (in the initial source token)
+        @param _minReturn       if the conversion results in an amount smaller than the minimum return - it is cancelled, must be nonzero
+        @param _toBlockchain    blockchain BNT will be issued on
+        @param _to              address/account on _toBlockchain to send the BNT to
+        @param _xTransferId     pre-determined unique (if non zero) id which refers to this transaction 
```

Comment:
I think we should also say something about the fact that the id allows matching between cross chain conversions, better than "id which refers to this transaction"